docs/data/data-grid/server-side-data/lazy-loading.md

```diff
@@ -2,14 +2,111 @@
+
+:::info
+Row count can be provided in either of the following ways.
+
+- Pass as [`rowCount`](/x/api/data-grid/data-grid/#data-grid-prop-rowCount) prop
+- Return `rowCount` in the `getRows` method of the [data source](/x/react-data-grid/server-side-data/#data-source)
+- Set the `rowCount` using the [`setRowCount`](/x/api/data-grid/grid-api/#grid-api-prop-setRowCount) API method.
+
+The above list is given in the order of precedence, which means if the row count is set using the API, that value gets overridden once a new value is returned by the `getRows` method, unless if it is `undefined`.
+:::
+
+## Viewport loading
+
+The viewport loading mode is enabled when the row count is known (`rowCount >= 0`). Grid fetches the first page immediately and adds skeleton rows to match the total row count. Other pages are fetched once the user starts scrolling and moves a skeleton row inside the rendering context (index range defined by [Virtualization](/x/react-data-grid/virtualization/)).
+
+If the user scrolls too fast, the grid loads multiple pages with one request (by adjusting `start` and `end` param) in order to reduce the server load.
+
+The demo below shows the viewport loading mode.
+
+{{"demo": "ServerSideLazyLoadingViewport.js", "bg": "inline"}}
+
+:::info
+The data source demos use a utility function `useMockServer` to simulate the server-side data fetching.
+In a real-world scenario, you should replace this with your own server-side data-fetching logic.
+
+Open info section of the browser console to see the requests being made and the data being fetched in response.
+:::
+
+### Request throttling
+
+While user is scrolling through the grid, rendering context changes and the Data Grid tries to fill in any missing rows by making a new server request.
+The Data Grid throttles new data fetches on the rendering context change to avoid doing unnecessary requests.
+The default throttle time is 500 milliseconds, use `unstable_lazyLoadingRequestThrottleMs` prop to customize it, as the following example demonstrates.
+
+{{"demo": "ServerSideLazyLoadingRequestThrottle.js", "bg": "inline"}}
+
+## Infinite loading
+
+The infinite loading mode is enabled when the row count is unknown (`-1` or `undefined`). New page is loaded when the scroll reaches the bottom of the viewport area.
+
+The area which triggers the new request can be changed using `scrollEndThreshold`.
+
+The demo below shows the infinite loading mode. Page size is set to `15` and the mock server is configured to return a total of `100` rows. Once the response does not contain any new rows, the grid stops requesting new data.
+
+{{"demo": "ServerSideLazyLoadingInfinite.js", "bg": "inline"}}
+
+## Updating the loading mode
+
+The grid changes the loading mode dynamically if the total row count gets updated in any of the three ways described above.
```

Comment:
```suggestion
The Grid changes the loading mode dynamically if the total row count gets updated in any of the three ways described above.
```

Reply:
Where and when were these "three ways" described above already? 😅 I've lost track by the time I've reached this point in the doc—maybe we can briefly reiterate them and/or link back to the section where they're described?

Reply:
I have reiterated "the ways"